### PR TITLE
First refactor in prep to kubeflow/kfctl/issues/36

### PR DESCRIPTION
**Pull request by nrchakradhar**
In preparation to [kfctl#36](https://github.com/kubeflow/kfctl/issues/36), I have refactored kfctl_go_test.py. Most of the refactoring is only about arrangement and making different steps in the test to functions.

- build_kfctl_go
- get_config_spec
-- Load Kubeflow spec, filter spartakus, modify repo
- Invoke kfctl init, generate, apply
- Verify

Based on feedback/comments, some code can be removed.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/4170)

### CODE REVIEW

File: testing/kfctl/kfctl_go_test.py [hunk 1]
```diff
@@ -147,32 +152,83 @@ def test_build_kfctl_go(app_path, project, use_basic_auth, use_istio, config_pat
...
+  #     to create the required K8s resources.
+  #  2. Setting the IAM policy will fail if the email is invalid.
+  email = util.run(["gcloud", "config", "get-value", "account"])
+
+  if not email:
+    raise ValueError("Could not determine GCP account being used.")
+
+  # username and password are passed as env vars and won't appear in the logs
+  #
+  config_spec = get_config_spec(config_path, project, email)
```
Comment (nrchakradhar) [inline testing/kfctl/kfctl_go_test.py]: Can rename to a better one. Suggestions please.

File: testing/kfctl/kfctl_go_test.py [hunk 2]
```diff
@@ -95,14 +83,17 @@ def test_build_kfctl_go(app_path, project, use_basic_auth, use_istio, config_pat
...
+        ".apps.googleusercontent.com")
 
   if use_istio:
     init_args.append("--use_istio")
   else:
     init_args.append("--use_istio=false")
 
+# Its not clear why this code is needed.
+# Will delete if approved.
+def dummy_version_repo_master():
```
Comment (jlewi) [inline testing/kfctl/kfctl_go_test.py]: I think this is obsolete. I think it was used in previous iterations when we were using a kfctl flag to specify the version of the repo to use.

File: testing/kfctl/kfctl_go_test.py [hunk 3]
```diff
@@ -147,32 +152,83 @@ def test_build_kfctl_go(app_path, project, use_basic_auth, use_istio, config_pat
...
+  kfctl_path = build_kfctl_go()
+  app_path, parent_dir = get_app_path_and_parent_dir(app_path)
+
+  logging.info("Using app path %s", app_path)
+  zone = 'us-central1-a'
+
+  # Set ENV for basic auth username/password.
+  set_env_init_args(use_basic_auth, use_istio)
+
+  dummy_version_repo_master()
```
Comment (jlewi) [inline testing/kfctl/kfctl_go_test.py]: I think you can delete this.

**Comment by jlewi** [comment]
/ok-to-test
Thanks for picking this up.

Just a couple minor comments.

File: testing/kfctl/kfctl_go_test.py (hunk 3 above)
Comment (nrchakradhar) [reply testing/kfctl/kfctl_go_test.py]: Deleted in the new commit. PTAL

**Pull request merged**